From 69135bcfe3c6e907e3dfed23d97eae9abfd782bc Mon Sep 17 00:00:00 2001
From: John Ralls <jralls@ceridwen.us>
Date: Tue, 11 Aug 2020 17:05:10 -0700
Subject: [PATCH] Accommodate macOS 11.0 Big Sur in gdk_quartz_osx_version.

This may need to be changed next year, it depends on what Apple
decides to do with version numbers going forward.

Also updated GdkOSXVersion to include Catalina and Big Sur.
---
 gdk/quartz/gdkglobals-quartz.c | 2 ++
 gdk/quartz/gdkquartz.h         | 4 +++-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/gdk/quartz/gdkglobals-quartz.c b/gdk/quartz/gdkglobals-quartz.c
index 46a4c811aa..72aed1c92d 100644
--- gdk/quartz/gdkglobals-quartz.c
+++ gdk/quartz/gdkglobals-quartz.c
@@ -42,6 +42,8 @@ gdk_quartz_osx_version (void)
 
       version = [[NSProcessInfo processInfo] operatingSystemVersion];
       minor = version.minorVersion;
+      if (version.majorVersion == 11)
+        minor += 16;
 #endif
     }
 
diff --git a/gdk/quartz/gdkquartz.h b/gdk/quartz/gdkquartz.h
index b6bce2b140..dd1b4f41af 100644
--- gdk/quartz/gdkquartz.h
+++ gdk/quartz/gdkquartz.h
@@ -40,7 +40,9 @@ typedef enum
   GDK_OSX_SIERRA = 12,
   GDK_OSX_HIGH_SIERRA = 13,
   GDK_OSX_MOJAVE = 14,
-  GDK_OSX_CURRENT = 14,
+  GDK_OSX_CATALINA = 15,
+  GDK_OSX_BIGSUR = 16,
+  GDK_OSX_CURRENT = 15,
   GDK_OSX_NEW = 99
 } GdkOSXVersion;
 
-- 
GitLab

